src/azul/service/repository_service.py

```diff
@@ -70,12 +72,60 @@ def __init__(self, entity_type: str, entity_id: str):
         super().__init__(f"Can't find an entity in {entity_type} with an uuid, {entity_id}.")
 
 
+@attrs.frozen(auto_attribs=True, kw_only=True)
 class SearchResponseStage(_ElasticsearchStage[ResponseTriple, MutableJSON],
                           metaclass=ABCMeta):
+    file_url_func: FileUrlFunc
+    filters: Filters
 
     def prepare_request(self, request: Search) -> Search:
         return request
 
+    @property
+    def repository_plugin(self) -> RepositoryPlugin:
+        return self.service.repository_plugin(self.catalog)
+
+    def inject_accessible(self, hits: JSONs):
+        special_fields = self.plugin.special_fields
+        for hit in hits:
+            entity = one(hit[self.entity_type])
+            source_id = one(hit['sources'])[special_fields.source_id]
+            entity[special_fields.accessible] = source_id in self.filters.source_ids
+
+    def inject_file_urls(self, node: AnyMutableJSON, path: list[str]):
```

Comment:
A tuple would also work here, right?
```suggestion
    def inject_file_urls(self, node: AnyMutableJSON, path: Sequence[str]):
```